test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -75,11 +75,14 @@ void test_ib_md::ib_md_umr_check(void *rkey_buffer,
     status = uct_md_mkey_pack(md(), memh, rkey_buffer);
     EXPECT_UCS_OK(status);
 
+    status = uct_md_mkey_pack(md(), memh, rkey_buffer);
+    EXPECT_UCS_OK(status);
+
 #ifdef HAVE_MLX5_HW
     uct_ib_md_t *ib_md = (uct_ib_md_t *)md();
 
     if (amo_access) {
-        if (check_umr(ib_md)) {
+        if (ib_md->relaxed_order || check_umr(ib_md)) {
```

Comment:
unite the 2 if's:
if ((amo_access &umr) || relaxed_order ) {
  expect 2nd key
} else {
 expect only 1 key
}